cmd/controller-manager/app/options/options.go

```diff
@@ -120,14 +121,27 @@ func (s *ServerOption) AddFlags(fs *pflag.FlagSet, knownControllers []string) {
 		"'-' to disable controllers, e.g. \"-job-controller,-queue-controller\" to disable job and queue controllers.", knownControllers))
 }
 
-// CheckOptionOrDie checks the option and returns error if it's invalid
+// CheckOptionOrDie checks all options and returns all errors if they are invalid.
+// If there are any invalid options, it aggregates all the errors and returns them.
 func (s *ServerOption) CheckOptionOrDie() error {
-	// check controllers option
+	var allErrors []error
+
+	// Check controllers option
 	if err := s.checkControllers(); err != nil {
```

Comment:
Before this pr, err will return directly once `checkControllers` has err, but after the change, other err will also be checked, which cause more code be executed, is this reasonable?

Reply:
IMO, this makes sense, because we are checking the options. The options should be checked as a whole and throw all errors, rather than throwing errors at each step.

Reply:
How about k8s's implements? Some time only the first error is need to be paid attention to.

Reply:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-scheduler/app/options/options.go#L267

We can refer to the implementation of kube-scheduler. The configuration is returned only after the entire verification is completed. The same goes for other components